indico/htdocs/js/indico/jquery/ajaxdialog.js

```diff
@@ -140,7 +139,9 @@
                 popup.canvas.on('ajaxDialog:reload', function() {
                     loadDialog();
                 });
+
                 injectJS(dialogData.js);
+                ajaxifyForms();
```

Comment:
The problem was that setting initial data on a form was executed before injecting JS code related to new widget. The library apparently is responsible for creating a hidden input, so in that case, the data from the field was not included in the form's initialData.